samples/database-create-with-encryption-key.js

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+async function createDatabaseWithEncryptionKey(
+  instanceId,
+  databaseId,
+  projectId,
+  keyName
+) {
+  // [START spanner_create_database_with_encryption_key]
+  // Imports the Google Cloud client library
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+  // const keyName =
+  //   'projects/my-project-id/my-region/keyRings/my-key-ring/cryptoKeys/my-key';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  // Gets a reference to a Cloud Spanner instance
+  const instance = spanner.instance(instanceId);
+
+  const request = {
+    encryptionConfig: {
+      kmsKeyName: keyName,
```

Comment:
Yeah this is confusing. But `CreateDatabaseRequest` has an `EncryptionConfig` field which only lets you specify the [`kms_key_name`](https://github.com/googleapis/googleapis/blob/8d17d8fafbb87ac64bb3179b99ac34ed41375a51/google/spanner/admin/database/v1/common.proto#L56).

Reply:
I think this might be because by default, it uses `GOOGLE_DEFAULT_ENCRYPTION`. But if the customer specifies an encryption key, then it has to be `CUSTOMER_MANAGED_ENCRYPTION`.